src/software/geom/util.cpp

```diff
@@ -1344,22 +1344,90 @@ std::vector<Circle> findOpenCircles(Rectangle rectangle, std::vector<Point> poin
     // on the field, and the size of the circle is the distance to the closest vertex
     // on the triangle that this vertex was created from
 
-    // TODO: instead of just adding the 4 corners of the rectangle here, the proper
-    //       thing to do is to generate the voronoi with just the points given, then
-    //       also treat all intersections of the voronoi diagram with the edge of the
-    //       rectangle as vertices
-    //       https://stackoverflow.com/questions/34093602/efficient-algorithm-to-determine-largest-open-space
-
-    // Add the 4 corners of the rectangle to the list of points
-    points.emplace_back(rectangle.posXPosYCorner());
-    points.emplace_back(rectangle.negXPosYCorner());
-    points.emplace_back(rectangle.posXNegYCorner());
-    points.emplace_back(rectangle.negXNegYCorner());
+    std::vector<Circle> empty_circles;
+
+    // Creating the Voronoi diagram with 2 or less points will produce no edges no we need
+    // to handle these cases manually
+    if (points.empty())
+    {
+        // If there are no points, return a circle encompassing the entire rectangle
+        empty_circles.emplace_back(Circle(
+            rectangle.centre(), (rectangle.negXNegYCorner() - rectangle.centre()).len()));
+        return empty_circles;
+    }
+    else if (points.size() == 1)
+    {
+        // If there is only 1 point, return a circle centered at the corner furthest from
+        // the provided point
+        Point furthestPoint = rectangle.furthestCorner(points.front());
+        empty_circles.emplace_back(
+            Circle(furthestPoint, (points.front() - furthestPoint).len()));
+        return empty_circles;
+    }
+    else if (points.size() == 2)
+    {
+        // If there are 2 point, split the points with a vector perpendicular to the
+        // vector connecting the two points. Return 2 circles that are centered at the
+        // points where the splitting vector intercepts the rectangle.
```

Comment:
Maybe I'm misinterpreting, but is it possible for the centre of the rectangle not to be covered by the 2 circles?